app/channel.go

```diff
@@ -27,6 +28,7 @@ import (
 // channelsWrapper provides an implementation of `product.ChannelService` to be used by products.
 type channelsWrapper struct {
 	srv *Server
+	app *App
```

Comment:
As I pointed in the other PR, I am not comfortable having both server and the app in the same struct, when server is accessible from app. We are starting to regress to the old style of no clear separation between app and server. @isacikgoz - can you create a ticket for this to fix it?

Reply:
Agreed. I was just trying to avoid a merge conflict, knowing it was be fixed.

Reply:
cc @gbochora